Add autogen for AMDC article in control with AMDC folder (#151)
Severson-Group/docs.amdc.dev: source/getting-started/control-with-amdc/auto-gen/index.md

1. **Input/Output (I/O):** Used for simulation and visualization only  
2. **Plant:** Represents the physical system (used for simulation)  
3. **Controller:** Contains the control logic to be deployed  

Only the **controller subsystem** is converted into embedded C code.

## Generated Code and Execution Model

Simulink Autogen produces C code that represents the controller as a callable function. This generated code should be treated as a black-box implementation of the Simulink controller. Within the AMDC, the control task is responsible for executing the controller at a fixed time interval. This fixed-time execution model is fundamental to digital control implementation on the AMDC. The execution sequence is:

1. Populate inputs using sampled sensor data  
2. Call the controller step function 
3. Route outputs to actuators (e.g., PWM duty cycles)  

The generated code has the following structure:

```c
void control_task_callback(void)
{
    // Populate inputs
    modelName_U.current = measured_current;
    modelName_U.voltage = measured_voltage;

    // Execute controller
    modelName_step();

    // Apply outputs
    set_pwm_duty(modelName_Y.duty);
}

> ```

Reviewer: Can we add each input, output and call function of autogen as code comments, as follows?

```suggestion
```c
void control_task_callback(void)
{
    // Populate inputs
    modelName_U.current = measured_current;  // Inputs to controller
    modelName_U.voltage = measured_voltage;  // Inputs to controller

    // Execute controller
    modelName_step();  // A function that executes the control algorithm

    // Apply outputs
    set_pwm_duty(modelName_Y.duty);  // Outputs from controller
}
```
```

- Input and output data structures:

```c
modelName_U   // Inputs to controller
modelName_Y   // Outputs from controller
```

- A function that executes the control algorithm:
  
```c
modelName_step();

> ```

Reviewer: If you agree to include my suggestion above, we do not need these lines below.
```suggestion
```

## Development Environment and Workflow

To develop control code using Simulink Autogen, the following software components are required:

- MATLAB  
- Simulink  
- Simulink Coder  
- Embedded Coder  

Additional toolboxes may be required depending on the specific control design.

### Recommended Workflow

The recommended workflow for developing control code is:

1. Develop and validate the control algorithm in Simulink  

> 2. Isolate the controller subsystem  

Reviewer: ```suggestion
2. Convert the controller to an atomic subsystem and then to a referenced model
```

3. Generate C code using Simulink Autogen  
4. Integrate generated code into the AMDC project  
5. Execute and validate on hardware  

### Repository Organization and Integration

> After code generation, Simulink creates a folder (typically named `modelName_ert_rtw`) that contains all generated source and header files. This folder must be incorporated into the AMDC project. A recommended approach is to place the generated folder within the application directory in the AMDC repository. For example:

Reviewer: It is actually enough to include only files, not the folder. I suggest the following:

```suggestion
After code generation, Simulink creates a folder (typically named `modelName_ert_rtw`) that contains all generated source and header files. To keep them separate from hand-written code for clarity and maintainability, all autogenerated `.c` and `.h` files generated files should be placed within the application directory in the AMDC repository. For example:
```

```
apps/
└── my_app/
    ├── src/
    ├── include/
    └── autogen/
        └── modelName_ert_rtw/
```

In this structure:

- The `modelName_ert_rtw` folder contains all autogenerated `.c` and `.h` files.  
- The folder is kept separate from hand-written code for clarity and maintainability.  

The autogenerated code should then be integrated into the AMDC project as follows:

- Add the `modelName_ert_rtw` directory to the compiler include paths.  

> - Add all generated `.c` files to the project source files.

Reviewer: These are now combined to the explanation above, and we do not need to add the `modelName_ert_rtw` directory as long as the user app is registered within the SDK.

```suggestion
```

Once added, the autogen folder should appear as part of the project within the SDK environment.

### Integration Notes

- Do not delete any generated files, as some auxiliary files may be required during compilation.
- Generated source (`*.c`) and header (`*.h`) files must be included in the AMDC project.
- The path to the folder containing the MATLAB script and the Simulink model should not contain any whitespace. If the folder or any parent directory contains whitespace in its name, the code generation process may result in a build error  

> This organization ensures that the autogenerated controller code is correctly compiled and integrated into the AMDC firmware.

Reviewer: Can we combine this section into the next section? For example, discussion about the whitespace issue is duplicated with the item 7 in the next section.

Author: That is good idea @noguchi-takahiro. I have revised it as you said. Thanks!
